src/components/Carousel.js

```diff
@@ -636,6 +638,9 @@ export default class Carousel extends Component {
    * @return {ReactElement} element
    */
   renderArrowRight = () => {
+    const { value, children, slides } = this.props;
+    const lastSlideIndex = (slides ? slides.length : React.Children.count(children)) - 1;
```

Comment:
https://github.com/brainhubeu/react-carousel/pull/125#discussion_r330561130

As said in the above comment, you should use `this.getCurrentValue` method for retrieving current value, and `this.getChildren` to get a list of slides.

Reply:
@udaypydi are you going to resolve this?

Reply:
@piotr-s-brainhub yeah i have resolved the issue. Will test it and update the PR tonight

Reply:
thanks, that will be great

Reply:
Yes, please! Waiting for this change :D

Reply:
@udaypydi There is no need for this ternary. Just calculate the last index of `slides` const here.